### PR TITLE
Updates for v1.2

### PR DESCRIPTION
**Pull request by danomatika**
This is a series of modernization updates and a couple features required by our usage at ZKM:

* bumped to version 1.2
* update the pd-lib-builder makefile
* added vbap loudspeaker set info outlet which reports speaker triplets and current triplet
* lowered loudspeaker intersection epsilon for float-based math (should this be settable?)
* limit min spread to avoid NaNs
* updated help files and removed pddp links for now
* commented our a few extraneous logposts and various formatting fixes
* updated changelog

### CODE REVIEW

File: define_loudspeakers.c [hunk 1]
```diff
@@ -560,7 +556,7 @@ static int lines_intersect(int i,int j,int k,int l,t_ls  lss[MAX_LS_AMOUNT])
   t_float dist_ij,dist_kl,dist_iv3,dist_jv3,dist_inv3,dist_jnv3;
   t_float dist_kv3,dist_lv3,dist_knv3,dist_lnv3;
   // TODO epsilon needs to be updated for 64-bit/double precision
-  t_float epsilon = 1e-9;
+  t_float epsilon = 0.01; // lower precison a bit for floats, was 1e-9;
```
Comment (umlaeute) [inline define_loudspeakers.c]: i don't know the implications of this, but it seems that the change is rather invasive.

File: rvbap.c [hunk 2]
```diff
@@ -21,6 +21,8 @@ See copyright in file with name LICENSE.txt  */
 #include "m_pd.h"               /* you must include this - it contains the external object's link to pure data */
 #endif
 
+#define RVBAP_VERSION "rvbap v1.2 - (c) Olaf Matthes 2003-2007, based on vbap by Ville Pulkki"
```
Comment (umlaeute) [inline rvbap.c:24]: nitpick, but i guess `RVBAP_VERSION` should be just `1.2` and the rest of the string should not be part of *this* define.

File: vbap.h [hunk 3]
```diff
@@ -28,24 +28,21 @@
 #define MAX_LS_AMOUNT 55    // maximum amount of loudspeakers, can be increased, but see comments next to MAX_LS_SETS above
 #define MIN_VOL_P_SIDE_LGTH 0.01  
 
-#define VBAP_VERSION "vbap - v1.1 - 14 Aug. 2014 - (c) Ville Pulkki 1999-2006 (Pd port by HCS)"
-#define DFLS_VERSION "define_loudspeakers - v1.0.3.2 - 20 Nov 2010 - (c) Ville Pulkki 1999-2006"
+#define VBAP_VERSION "vbap v1.2 - 17 Jul 2018 - (c) Ville Pulkki 1999-2006 (Pd port by HCS)"
+#define DFLS_VERSION "define_loudspeakers v1.2 - 17 July 2018 - (c) Ville Pulkki 1999-2006"
```
Comment (umlaeute) [inline vbap.h:32]: same here. I'd go and define a single `VBAP_VERSION=1.2` and use that... (btw, where are the dates coming from?)

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[submodule "pd-lib-builder"]
```
Comment (umlaeute) [inline .gitmodules]: i'd rather use `git subtree` than `git submodule`. it makes it much easier to build from a "release tarball" (as opposed to to "git clone")

**Review by umlaeute — COMMENTED** [review]
not much to say :-)
the only thing i'd really suggest is to switch to git-subtree for the pd-lib-builder.

File: define_loudspeakers.c (hunk 1 above)
Comment (danomatika) [reply define_loudspeakers.c]: Yeah, I'm sure how necessary this really is but I didn't wan to "jinx" a running system.

Comment (danomatika) [reply define_loudspeakers.c]: This might be better as an overridable define or event a run time message.

File: vbap.h (hunk 3 above)
Comment (danomatika) [reply vbap.h:32]: Oh, that's when I did the actual work last summer. This is post-release cleanup.

File: .gitmodules (hunk 4 above)
Comment (danomatika) [reply .gitmodules]: Ok.

File: vbap.h (hunk 3 above)
Comment (danomatika) [reply vbap.h:32]: My reasoning was: "I am a user and I load this object. I expect to see the date when the project was last updated, not the last attribution date."

File: rvbap.c (hunk 2 above)
Comment (danomatika) [reply rvbap.c:24]: Eh, I'm not going to change these as they are only used internally. If we add a correct `VBAP_VERSION` define for a `VBAP_SETUP_MSG` define string, it's only used there anyway...

Comment (danomatika) [reply rvbap.c:24]: I'm also letting the terrible formatting jumbled these sources go as well...

File: .gitmodules (hunk 4 above)
Comment (danomatika) [reply .gitmodules]: I may just include the lib builder directly as with earplug~. Easy enough for just 1 file.

File: define_loudspeakers.c (hunk 1 above)
Comment (danomatika) [reply define_loudspeakers.c]: Now using FLOAT_EPSILON.

File: .gitmodules (hunk 4 above)
Comment (umlaeute) [reply .gitmodules]: `git subtree` makes it easier to upgrade: keeping history *and* not having to worry about whether the it stays single-file.

anyhow: just make sure to use `PDLIBBUILDER_DIR`

Comment (danomatika) [reply .gitmodules]: Yup and we should update the pdlibbuilder hello-world example to use it as well. That's what I copy/pasted yesterday...

**Comment by danomatika** [comment]
At this point, I'm still not sure what the difference between `[vbap]` and `[rvbap]` is.

File: .gitmodules (hunk 4 above)
Comment (umlaeute) [reply .gitmodules]: you mean like https://github.com/pure-data/helloworld/commit/dd1554035fcf9992a4c887853a682c7c13d5454c? (note the date)

Comment (umlaeute) [reply .gitmodules]: ah, by *it* did you mean `git subtree` (rather than `PDLIBBUILDER_DIR`, as i assumed)?

Comment (danomatika) [reply .gitmodules]: No, I guess I used on old cloned copy of helloworld.

Comment (danomatika) [reply .gitmodules]: In looking at this again, I already set it up as a subtree and now I remember that I find subtrees confusing. In this case, it's a single file to update and I don't want to pull in all the lib builder tests with it.

**Comment by danomatika** [comment]
Additional question: should the new `[vbap]` right outlet be optional and created via a creation arg or do we just create it by default? This outlet sends out the loaded speaker triplets as lists and the current triplet as the position is changed. We use this in our project to both render the speaker triplet mesh and update which triangle is active per-sound ID.

**Comment by danomatika** [comment]
I've gone ahead and created a new branch `update/1.2.0` by cherry-picking the non subtree stuff.

**Comment by danomatika** [comment]
Superseded by #2.

**Pull request closed without merging**